grader_service/handlers/base_handler.py

```diff
@@ -986,21 +986,32 @@ def _run_command(self, command, cwd=None, capture_output=False):
             self.log.error(e)
             raise HTTPError(404, reason="File not found")
 
-    async def _run_command_async(self, command, cwd=None):
-        """Starts a sub process and runs a cmd command
+    async def _run_command_async(self, command_args: List[str], cwd=None):
```

Comment:
Now that I think of it, maybe it's better to pass the whole command as a string (like we've been doing to far), and use [`shlex.split`](https://docs.python.org/3/library/shlex.html#shlex.split) to convert it into a list? I'm not sure if this is a better solution in itself, but we also do it in the `_run_command` method (above this one in the file, lines 973-985), so we could change it for consistency's sake.